lib/replica_pools/query_cache.rb

```diff
@@ -20,12 +20,15 @@ def #{method_name}(*a, &b)
     # select_all is trickier. it needs to use the leader
     # connection for cache logic, but ultimately pass its query
     # through to whatever connection is current.
-    def select_all(arel, name = nil, binds = [])
+    def select_all(*args)
+      # there are more args, but we only care about arel and binds for caching
+      arel, _, binds = args
       if query_cache_enabled && !locked?(arel)
         sql = to_sql(arel, binds)
-        cache_sql(sql, binds) { route_to(current, :select_all, sql, name, binds) }
+        args[0] = sql
+        cache_sql(sql, binds) { route_to(current, :select_all, *args) }
```

Comment:
Looks like travis is picking up Rails 5.1, which is failing due to the new [`cache_sql` signature](https://github.com/rails/rails/blob/v5.1.0/activerecord/lib/active_record/connection_adapters/abstract/query_cache.rb#L103).

I can fix up our test suite to run on specific rails versions in a followup PR — for now, would you mind adding a conditional here for Rails 5.1 support? Should be easy fix 😅